### ISSUE TITLE
 Repeat affixed

### ISSUE THREAD

**Issue by chenzengpeter** [arduino/Arduino#7061]
I am using arduino on mac but will paste it again when I copy it.

**Comment by per1234** [comment]
It's unclear what you're saying. You need to provide much more detail.

**Comment by chenzengpeter** [comment]
if i (command c) ---> copy  print
but (command v) ---> Posted on the print will be repeated affixed twice  It becomes printprint

**Comment by per1234** [comment]
Which version of the Arduino IDE are you using? Please test with the [beta build](https://www.arduino.cc/en/Main/Software#beta) to verify the problem has not already been fixed.

I seem to remember this having been reported previously but I can't find it. I actually did have this same problem for a while but haven't for a long time and I can't remember what changed. I do have a keyboard remapping of my copy/paste shortcuts and changed the application I use to do that at one point so it may have been caused by a bug in the previous application.

**Comment by chenzengpeter** [comment]
I currently use Arduino 1.8.5
I tested the problem still exists
And then you see what you say changed the shortcut key inside the thing
I would like to change (command v) to (control v or option v) one of the shortcut keys
Fantastic thing happened (command v) can be used normally  but press (control v or option v) can not be used
A magical bug
I remember when I bought it (command v) can be used normally to update from a mac version to die 
i buy mac in 2014 year
thank you for your help

**Comment by lukec** [comment]
This bug happens when your keyboard input is set to anything other than US English. It's easy to reproduce on a mac:

1. set your keyboard input to a different locale (eg: Dvorak or other languages)
2. open any sketch 
3. copy some text
3. paste (Cmd-V) into the sketch.

Expected: pasting "foo"

> foo

Actual:

> foofoo

**Comment by Sidchou** [comment]
same problem, basically the txt editor read my short cut key twice when  i do it once, also happen with Cmd+/, it comment out and then back the line

**Comment by yakeer** [comment]
+1

**Comment by NeilFraser** [comment]
This bug dates back to at least April 2015, version 1.6.3:
https://forum.arduino.cc/index.php?topic=315092.0

It is *extremely* irritating.  Of course since it mostly affects non-Americans, I guess it's pretty low on the priority list...

**Comment by gweep** [comment]
I have this problem still.... here is my solution..   PASTE (cmd-v)  gets you two copies.... UNDO (cmd-z) removes the 2nd copy only....   inelegant but it works for me.

**Comment by airs0urce** [comment]
I have same issue

**Comment by wtrout** [comment]
I'm a DVORAK user who just switched to mac. Dang this bug is annoying.

**Comment by auxiliaryjoel** [comment]
does anyone have a soluation for this. I screenshot items to my clipboard constantly as part of training for my clients,
in the past pasting this content behaved as it should - it would only paste the screenshot once into my gmail emails.
But in the last month or so that has changes to now always pasting the screenshots 2 at a time. 
What on earth has changed? (Mac OS 10.14.6 on Imac)

**Comment by prog666** [comment]
same for me, and I thought it has nothing to do with Dvorak I use because I tried Russian to check - but appears they are both broken...

**Comment by mertnesvat** [comment]
yep it's related to the keyboard layout in Dvorak it's double pasting but USA keyboard works fine.

**Comment by gweep** [comment]
Holy cow!  How long is this going to remain unaddressed?

**Comment by per1234** [comment]
There is a good chance it already has been addressed. Please check to see whether it still occurs with the latest nightly build of Arduino IDE 2.x:

https://www.arduino.cc/en/software#nightly-builds

**Comment by zaneclaes** [comment]
@per1234 if you look just above your post, you'll see that someone tested the nightly build just 4 days before your post. Some of us have been [repeatedly reporting this issue for over 3 years now](https://github.com/arduino/Arduino/issues/10534). 

The _problem has actually gotten MUCH worse after the latest nightly build and the switch to Arduino IDE_. Now, the hotkeys do something completely different. For example, if I press `CMD+C` (using Dvorak+QWERTY) to copy something, it instead makes the Output tab disappear (and does not copy anything). If I press `CMD+V` (to paste), it says `CMD+K was pressed, waiting for more keys` (and again does not actually do anything).
